.travis.yml

```diff
@@ -0,0 +1,42 @@
+services:
+  - docker
+language: ruby
+before_install:
+  - rm Gemfile.lock || true
+  - gem update bundler
+  - gem --version
+  - bundler -v
+  - 'if [ -n "$DOCKER_USERNAME" -a -n "$DOCKER_PASSWORD" ]; then echo "$DOCKER_PASSWORD" | docker login -u "$DOCKER_USERNAME" --password-stdin ; fi'
+stages:
+  - test
+  - build
+  - name: deploy
+    if: branch = main OR tag IS present
```

Comment:
There is a lot of conditional logic in here. Could you please add a `CONTRIBUTING.md` that explains how the contribution process works and what triggers new containers to be published.

Reply:
Added `CONTRIBUTING.md` and explained docker image processes.